api-get-object.go

```diff
@@ -289,6 +292,21 @@ func (o *Object) doGetRequest(request getRequest) (getResponse, error) {
 		return response, response.Error
 	}
 
+	// Check if ETag has changed since last request.
+	switch {
+	// If response doesn't carry a valid ETag, ignore.
+	case response.objectInfo.ETag == "":
+
+	// Set Object's etag on first request.
+	case o.etag == "":
+		o.etag = response.objectInfo.ETag
+
+	// Check if etag from response matches Object's etag.
+	case o.etag != response.objectInfo.ETag:
+		return response, syscall.EBADF
```

Comment:
It should be encapsulated inside ErrInvalidObjectNamet() @krisis which provides ErrorResponse compatible error.  This way caller can look at the `errResp.Code == NoSuchKey` and can decide to re-open the connection to discard and start reading again.